common/achievement/get.ts

```diff
@@ -0,0 +1,214 @@
+    const achievements = await Promise.all(
+        groupKeys.map(async (key) => {
+            const group = groups[key];
+            const sortedGroupAchievements = group.sort((a, b) => a.groupOrder - b.groupOrder);
+            if (sortedGroupAchievements[0].template.type === achievement_type_enum.TIERED) {
+                return await Promise.all(
+                    sortedGroupAchievements.map(async (groupAchievement) => {
+                        const achievement: Achievement = await assembleAchievementData([groupAchievement], user);
+                        return achievement;
+                    })
+                );
+            }
+            const groupAchievement: Achievement = await assembleAchievementData(sortedGroupAchievements, user);
+            return [groupAchievement];
+        })
+    );
+    return achievements.flat();
+};
+
+const assembleAchievementData = async (userAchievements: User_achievement[], user: User): Promise<Achievement> => {
+    let currentAchievementIndex = userAchievements.findIndex((ua) => !ua.achievedAt);
+    currentAchievementIndex = currentAchievementIndex >= 0 ? currentAchievementIndex : userAchievements.length - 1;
+
+    const achievementContext = transformPrismaJson(user, userAchievements[currentAchievementIndex].context);
+    const currentAchievementTemplate = getCurrentAchievementTemplateWithContext(userAchievements[currentAchievementIndex], achievementContext);
+
+    const achievementTemplates = await prisma.achievement_template.findMany({
+        where: { group: currentAchievementTemplate.group },
+        orderBy: { groupOrder: 'asc' },
+    });
+
+    const state: achievement_state = getAchievementState(userAchievements, currentAchievementIndex);
+
+    const isNewAchievement = state === achievement_state.COMPLETED && !userAchievements[currentAchievementIndex].isSeen;
+
+    const condition = currentAchievementTemplate.condition.includes('recordValue')
+        ? currentAchievementTemplate.condition.replace('recordValue', (userAchievements[currentAchievementIndex].recordValue + 1).toString())
+        : currentAchievementTemplate.condition;
+
+    let maxValue;
+    let currentValue;
+    if (currentAchievementTemplate.type === achievement_type_enum.STREAK || currentAchievementTemplate.type === achievement_type_enum.TIERED) {
+        const dataAggregationKeys = Object.keys(currentAchievementTemplate.conditionDataAggregations);
+        const relation = userAchievements[currentAchievementIndex].context['relation'] || null;
+        const evaluationResult = await evaluateAchievement(
+            user.userID,
+            condition,
+            currentAchievementTemplate.conditionDataAggregations as ConditionDataAggregations,
+            userAchievements[currentAchievementIndex].recordValue,
+            relation
+        );
+        currentValue = dataAggregationKeys.map((key) => evaluationResult.resultObject[key]).reduce((a, b) => a + b, 0);
```

Comment:
I think this doesn't work as the resultObject is including the recordValue as well.

Reply:
this is why data aggregation keys are use of which none is revordValue, therefore having record value being ignored here.